src/lib/docs/Utilities/SigningRequest.md

```diff
@@ -63,156 +84,219 @@ const request = SigningRequest.create({
 
-#### Options
+```typescript
+import { SigningRequest } from "@wharfkit/signing-request"
 
-- `scope` - The account name of the identity to be requested. Defaults to a placeholder value.
-- `permission` - The permission of the identity to be requested. Defaults to a placeholder value.
+const request = SigningRequest.identity({
+  callback,
+  permission: "account@active",
+})
+```
 
 ## Usage
 
-### Serialize the request into a URI:
+### Encoding a Request
 
-Once the request is created, it can be encoded as a URI string that can be passed onto wallets for signing:
+Once a `SigningRequest` instance is created, it can be encoded as a string in URI format that allows it to easily be passed to other applications:
 
 ```typescript
+import { SigningRequest } from "@wharfkit/signing-request"
+
+const request = SigningRequest.identity({
+  callback,
+  permission: "account@active",
+})
+
 const uri = request.encode()
+
 // alternatively:
 // const uri = String(request)
 ```
 
-A boolean can be passed to the `encode` method to specify whether the URI should be compressed:
+The `encode` method accepts 3 optional parameters:
 
-```typescript
-const uri = request.encode(false)
+- 1st: A boolean to specify whether the URI should be compressed
+- 2nd: A boolean to specify whether slashes should be included in the URI
+- 3rd: A string indicating what URI scheme should be used
 
-console.log(uri) // esr://gmNgZGRkAIFXBqEFopc6760yugsVYWBggtKCMIEFRnclpF9eTWUACgAA
+```ts
+request.encode(compress, slashes, scheme)
 ```
 
-The slashes can also be omitted and the scheme can be customized like this:
+### Decoding a Request
 
-```typescript
-const uri = request.encode(undefined, false, "custom-scheme")
+A URI string can be decoded into a `SigningRequest` instance using the `from` static method.
 
-console.log(uri) // custom-scheme:gmNgZGRkAIFXBqEFopc6760yugsVYWBggtKCMIEFRnclpF9eTWUACgAA
+```typescript
+const request = SigningRequest.from(
+  "esr://gmNgZGRkAIFXBqEFopc6760yugsVYWBggtKCMIEFRnclpF9eTWUACgAA"
+)
 ```
 
-### Decode a URI into a request:
+Once decoded, the `SigningRequest` instance will then be usable with any of the additional methods outlined below.
 
-A URI string can be decoded into a `SigningRequest` instance:
+### Resolving a Request
 
-```typescript
-const request = SigningRequest.from(uri)
-```
+A `SigningRequest` instance does not need always contain a transaction ready for signing or use on the blockchain. A signing request may be a single action, or an array of actions, or perhaps an action with placeholder values for the authorization or action data. For this reason, signing requests need to be resolved from this incomplete state into a resolved request.
+
+Resolving a signing request requires 3 pieces of information:
 
-### Resolving a Signing Request
+- The `abis` related to the actions in the request, to serialize the required data.
```

Comment:
In the case of identity requests, this parameter can be an empty array, right? Should we mention something about that?

Reply:
We could, though you can still go through the process I believe.

Reply:
Yeah it looks like in desktop its still being used, even on identity requests, despite probably be an empty array or something. 

https://github.com/greymass/anchor/blob/5444c52aeb6d4ff6a9e900e9bdf6b2303a9ab2a9/app/modules/handler/actions/uri.js#L565-L575

This is probably an area where we just need to revamp how ESR works.

Reply:
It looks like in the case of identity requests, the abi is synthesized so you could probably pass an empty array - https://github.com/wharfkit/signing-request/blob/master/src/signing-request.ts#L739

Reply:
I feel like with the way that it's phrased right now, readers are gonna think that they always need to pass abis 😅